### DISCUSSION TITLE
Livewire v3 wire:ignore not working

### DISCUSSION THREAD

**Discussion by schonhoff** [livewire/livewire#5813] [7. Bugs (archived)]
Hello,

I used some external javascript which is loading after Livewire init. In Livewire v2 I could just add `wire:ignore` to the parent node and it worked. Now it doesn't work and the documentation for Livewire v3 doesn't mention anything about `wire:ignore`. Was it removed? Is there another way for me to handle the situation?

The script I want to add is the google recaptcha.

**Comment by nuernbergerA** [comment]
`wire:ignore` is still a thing
maybe the fact that livewire uses alpine now it has some side-effects
try to use also [x-ignore](https://alpinejs.dev/directives/ignore)

**Reply by schonhoff** [reply]
Will try it out, thank you!

**Reply by vincentdev15** [reply]
Livewire also used AlpineJS in version 2.x.

**Reply by Kotzilla** [reply]
x-ignore not working either

**Comment by JeffGepiga** [comment]
If you have a lot of element that has if else condition, try to wrap it with wire:key each section like below

```
<div class="row">
<div class="col-sm" wire:key="col-1">
</div>
@if(condition)
<div class="col-sm" wire:key="col-2">
</div>
@endif
</div>
```

**Comment by moazam1** [comment]
I am having same issue. wire:ignore is not working after upgrade.

**Reply by Xety** [reply]
Same for me, I just upgraded to v3 and `wire:ignore` is... ignored

**Reply by moazam1** [reply]
Great. I am not alone. I switched back to v2.

**Reply by Xety** [reply]
Look like I will do the same since I have a Calendar displayed with `wire:ignore`, but the Calendar disapear when I trigger some creating/updating events...
Sad because everything seems to work well in my application after migrating expect that... 😢

**Comment by Xety** [comment]
Can we have any update about this bug from the team please ?

**Comment by bekovski** [comment]
I'm experiencing the same bug after upgrading to v3. Does anyone have a solution?

**Comment by fusionconsulting** [comment]
Same here. Causing load errors for elements that should be ignored

**Comment by bekovski** [comment]
Would you know if there is any priority for fixing the bug in wire:ignore?

**Comment by viliusvsx** [comment]
Try to use `wire:ignore.self`. For me this attribute is working, I am using this with bootstrap modal

**Comment by PhiloNL** [comment]
Please share some examples with easy-to-reproduce steps.
I just created an example and it works just fine:
https://wirebox.app/b/4qj2x

As you can see, the component refreshes every three seconds.
The timestamp is therefore updated every three seconds as well.
The quote listed below is not updated because of the `wire:ignore` directive.

**Reply by bekovski** [reply]
I copied your example to my project and the following bug happens: The first time you run wire:poll.3s it updates the message but then runs wire:ignore correctly.

**Reply by marttheis** [reply]
It doesn't work at all when you use alpineJS in combination wirth livewire3. Livewire always transfers the whole alpineJS component which means that you can't use alpinejs at all. We have a complex modal where every input is x-modelled. Alle input are then synced with Livewire3 on the outermost div. As a result livewire transfers the whole 90kb mark-up of the modal on any tiny input field value change... This is unbearable.

**Comment by jadayash** [comment]
@PhiloNL 
Am facing same issue. This is happening when the website is behind cloudflare. i switch my website to DNS only on cloudflare and the issue was solved.

**Reply by PhiloNL** [reply]
Make sure you disable HTML minification when using the proxy. This could cause issues with Livewire’s morphing system.

**Reply by jadayash** [reply]
100%, this was the issue
Auto Minify for HTML

**Reply by sirvan-monfared** [reply]
hi, i have the same issue. i dont know how to disable HTML minification. i am in local development with laragon. please help me

**Reply by jadayash** [reply]
![html](https://github.com/livewire/livewire/assets/66420205/fb7eb3c7-3b85-428f-9388-34fa18c1c526)

**Comment by Kotzilla** [comment]
Any Update on this issue. now im fixed with re init with `x-init` everytime on update/create

**Comment by marttheis** [comment]
I am experiencing major issues with my setup. I have a large and complex modal controlled by various variables, which are entangled with Livewire on the parent div of the frontend blade. The individual input fields are all using x-model, and I am not using any wire:model. Livewire 3 is only connected via $wire.entangle() in the parent div.

All inputs are interconnected, so when I change one, all the others are recalculated. However, with every Livewire update, the entire modal markup is being transferred, which is about 140kb each time.

I believe this might be a major bug in how AlpineJS interacts with Livewire 3. If your entire Livewire component is "alpinized," Livewire—for some reason—transfers the entire markup on each update. I have tried using wire:ignore, wire:ignore-self, and wire:key, but none of these methods seem to help.

Nothing in the frontend visibly changes during these Livewire updates, except for new values in the input fields. Yet, Livewire still swaps the entire markup every time.

This severely impacts performance and responsiveness, as we can't optimize transfer sizes properly.

**Reply by PhiloNL** [reply]
@marttheis if you could submit a failing test of this or provide code to replicate this specific issue that would be great.

**Reply by marttheis** [reply]
Yes I will provide an example tomorrow.

**Reply by marttheis** [reply]
In the meantime, I've tried two different approaches, one using $wire.set with this.$watch and another using @js with @this.set and this.watch.

I'm honestly not sure what else to try. This can't be how Livewire 3 is supposed to function. AlpineJS handles everything on the frontend, so why is Livewire still swapping the entire markup?

If this behavior is intentional and unavoidable, we may have to abandon Livewire 3. Our app relies heavily on sliders and input field manipulations, and this would cause massive payload sizes that could severely impact performance.

**Reply by marttheis** [reply]
I believe that the DOM diffing and subsequent morphing of livewire3 is fully removed whenever you use Alpine inside a livewire3 component. I modified my code so that no elements are added or removed ever. I got rid of x-cloak and x-show and the x-for loop. However Livewire3 still is unable to morph the DOM properly... it simply swaps the entire mark-up. 

https://livewire.laravel.com/docs/morphing

I am getting very frustrated.

Edit1: Here is my assumption: Livewire3 adds wire:replace on every elements which contains an x-data and all of it's children.

Edit2: Maybe I can somehow use the alpine morph plugin? Any idea on how to do that? https://alpinejs.dev/plugins/morph

Edit3: It seems that the only solution would be to go bonkers with nesting livewire components. So in our example it would mean to make the inputfield and the dropdown it's own livewire component and then rely on the mentioned ways of communicating between these components as explained here: https://livewire.laravel.com/docs/understanding-nesting#performance-implications

**Comment by marttheis** [comment]
As promised, here is an example of how we use Livewire 3 in our development. It's a straightforward approach without many hacks or workarounds.

I’ve attached the full frontend view of a simple input search box with buttons. Everything on the front end is powered by AlpineJS, except for the main div, which has all the entangled Livewire 3 properties.

The entire frontend markup is 12KB, and it gets transferred fully on every Livewire update. The actual JSON payload is tiny in comparison. I’d love to simply use wire:ignore.self on the heavier divs (which have x-transitions), but it doesn't seem to work. wire:ignore is completely disregarded, no matter where I place it in the markup. The transfer size remains the same on every Livewire update.

I’m not an experienced coder, so I’m unsure how a good example should look. However, I wanted to show you how we heavily utilize AlpineJS with stores, Alpine
(though not used here), and more.

Maybe we’re doing something wrong, or maybe the issue in our code is obvious? I’m not sure, but it’s becoming quite frustrating. It feels like we’re in a tough spot with our decision to use the TALL stack for this project when every small roundtrip transfers the entire markup of the Livewire component.

Any help would be greatly appreciated!

**Here is the livewire3 php blade:** 
```
<?php

namespace App\Livewire\Dashboard;
use Livewire\Component;


class CompSymbolSearchSingle extends Component
{
    public $searchSymbol;
    public $selectedSymbol;
    public $symbols;   
    public $tradeOpen; //check if a trade is already open or not
    
    public function mount()
    {
        $this->searchSymbol ='';
        $this->selectedSymbol = '';
        $this->symbols = [];
        $this->tradeOpen = false; 
    }

    public function updatedSearchSymbol($searchSymbol)
    {
       
    }


    public function updatedSelectedSymbol($selectedSymbol)
    {
     
    }


    public function render()
    {
        return view('dashboard.comp-symbol-search-single');
    }
}

```

**And here our general frontend approach:** 

```
<div>
    <div 
        class="tab:padtop pt-5" 
        name="symbolsearch" 
        id="symbolsearch" 

        x-data="{
            searchSymbol: $wire.entangle('searchSymbol').live,
            symbols: $wire.entangle('symbols').live || [], 
            selectedSymbol: $wire.entangle('selectedSymbol').live,
            tradeOpen: $wire.entangle('tradeOpen').live,
            

            openDropdown: false,
            open: false,

            init() {
               
                this.$watch('symbols', (newSymbols) => {
                    this.openDropdown = newSymbols.length > 0;
                });
            }
        }">

            <div class="relative flex justify-center ">

                <div class="flex flex-row items-center justify-center w-full gap-2 text-tsec tab:w-3/5 desk:w-2/5">

                    <div  x-cloak  x-transition:enter="transition ease-out duration-[220ms]"
                        x-transition:enter-start="opacity-0  translate-x-[-50px]" x-transition:enter-end="opacity-100 "
                        x-transition:leave="transition ease-in duration-[50ms] " x-transition:leave-start="opacity-80 "
                        x-transition:leave-end="opacity-0  translate-x-[-10px]" x-show="open"
                        class="flex items-center justify-center transition-all transform-gpu w-fit">
                        
                        
                        <button type="button" @click=" $modals.show('create-new-position'); Alpine.store('setNewPositionModal').setSymbolAndDirection(selectedSymbol, 1)"
                       :disabled="tradeOpen"
                            tabindex="0"
                            class="w-fit tab:w-[130px] h-14 flex items-center justify-center gap-2 touch-active rounded-lg   "
                            :class="{'button-idle': tradeOpen, 'button-p1': !tradeOpen}"><span
                                class="hidden tab:flex">Long</span><svg class="h-5 " viewBox="-1 0 26 24" fill="none"
                                stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round">
                                <polyline points="23 6 13.5 15.5 8.5 10.5 1 18"></polyline>
                                <polyline points="17 6 23 6 23 12"></polyline>
                            </svg>
                        </button>



                    </div>

                    <div  class="flex items-center flex-grow tab:flex-grow-0 tab:min-w-full input-icon input-icon-right justify-self-center">
                    
                        <div   x-show="!open" class="h-full input-icon-search"  
                            x-transition:enter="transition ease-out duration-[600ms]"
                            x-transition:enter-start="opacity-0 " 
                            x-transition:enter-end="opacity-100"
                            style="transition-delay: 150ms;"> 
                            <svg  
                            class="will-change-transform h-6 w-6 text-tsec2 stroke-[2]" 
                            viewBox="0 0 24 24" fill="none" stroke="currentColor" 
                            stroke-linecap="round" stroke-linejoin="round"
                            >
                            <circle cx="11" cy="11" r="8"></circle>
                            <line x1="21" y1="21" x2="16.65" y2="16.65"></line>
                            </svg> 
                        
                            <div class="text-tsec" >Open new position</div>

                        </div>
                    

                        <input 
                            x-ref="inputComponent" 
                            x-model.debounce.150ms="searchSymbol"
                            x-on:input="open = $event.target.value.length > 0" 
                            x-on:updated="open = false"
                            :aria-expanded="open ? 'true' : 'false'" 
                            :class="[
                                open ? '!pl-4' : '!pl-[52px]',
                                {'active': open},
                               
                            ]"
                            class="placeholder:text-tsec2  b1 text-tpri bg-e1 flex justify-start h-14 w-full items-center rounded-lg"
                            id="symbol-input"
                            onkeydown="return (event.keyCode!=13);"  
                            placeholder=""
                            type="text" 
                            size="6"
                            autocomplete="off"
                            autocorrect="off" 
                            autocapitalize="off" 
                            spellcheck="false"
                        />
                            <i x-cloak x-show="open" 
                            class="!h-14 !w-14 !bg-transparent transition-all transform-gpu">
                                <button  x-cloak 
                                        x-on:click="searchSymbol = '';  $refs.inputComponent.blur(); $refs.inputComponent.dispatchEvent(new Event('updated'))"
                                        type="button" 
                                        id="symbol-toggle" 
                                        class="flex items-center justify-center w-full h-full text-tsec active:text-s1 ">
                                    <svg id="symbol-visible" 
                                        class="h-6 cursor-pointer stroke-2" 
                                        xmlns="http://www.w3.org/2000/svg" 
                                        fill="none" 
                                        viewBox="0 0 24 24" 
                                        stroke="currentColor">
                                        <path stroke-linecap="round" stroke-linejoin="round" d="M6 18L18 6M6 6l12 12" />
                                    </svg>
                                </button>
                            </i>
                    </div>

                    <div  x-cloak  id="showShortButton"
                        x-transition:enter="transition ease-out duration-[220ms]"
                        x-transition:enter-start="opacity-0  translate-x-[50px]" x-transition:enter-end="opacity-100"
                        x-transition:leave="transition ease-in duration-[50ms] " x-transition:leave-start="opacity-80"
                        x-transition:leave-end="opacity-0  translate-x-[10px]" x-show="open"
                        class="flex items-center justify-center transition-all transform-gpu w-fit ">
                        <button :disabled="tradeOpen" type="button" @click=" $modals.show('create-new-position'); Alpine.store('setNewPositionModal').setSymbolAndDirection(selectedSymbol, 0)"
                             tabindex="0"
                            class="w-fit tab:w-[130px] h-14 flex items-center justify-center gap-2  touch-active rounded-lg "
                            :class="{'button-idle': tradeOpen, 'button-s1': !tradeOpen}">
                            <div class="hidden tab:flex">Short</div><svg class="h-5 " viewBox="-1 0 26 24" fill="none"
                                stroke="currentColor" stroke-width="3" stroke-linecap="round" stroke-linejoin="round">
                                <polyline points="23 18 13.5 8.5 8.5 13.5 1 6"></polyline>
                                <polyline points="17 18 23 18 23 12"></polyline>
                            </svg>
                        </button>
                    </div>

                </div>
            

        

                <div x-cloak x-show="openDropdown" id="symbol-list"
                    class="absolute shadowe4 tab:top-16 top-16 z-[99] bg-e1 w-full tab:w-3/5 desk:w-2/5 rounded-lg">

                    <ul class="pt-2 pb-2 overflow-hidden">
                        <template x-for="symbol in symbols" :key="symbol.Symbol">
                            <li class=" flex flex-1 h-fit border-b border-table last:border-none">
                                <a @click="window.scrollTo({top: 0, behavior: 'smooth'}); openDropdown = false; selectedSymbol = symbol.Symbol"
                                    class="w-full h-full px-4 py-1 tab:py-2 hover:none desk:hover:bg-exborder active:none desk:active:bg-tainput iphone">
                                    <div class="flex flex-row justify-between">
                                        <span class="font-bold text-tpri2" x-text="symbol.Symbol"></span>
                                        <span class="text-tpri b2" x-text="symbol.StockExchange"></span>
                                    </div>
                                    <div class="flex flex-row justify-between b2">
                                        <span class="text-tsec2 pr-1 b2 overflow-hidden whitespace-nowrap text-ellipsis" x-text="symbol.CompanyName"></span>
                                        <span class="text-tsec2 b2">stock</span>
                                    </div>
                                </a>
                            </li>
                        </template>
                    </ul>
                </div>

            </div>

    </div><!--Ende main x-data -->  
</div><!--Ende empty livewire DIV -->


```

**Comment by gmartinez** [comment]
Same here...after 2-3 days working around moving to v3, having been able to solve all issues I started tripping on...I finally got to this wall I could not go through, ended up reverting to v2...I'll keep trying in its own branch, I'll monitor this thread I guess.

I wish I could bring here little more than a +1, but I have no simple vanilla case to demonstrate, I'll try to provide one.

**Comment by marttheis** [comment]
Alright guys I found a  working solution to my problem. 
I now have all my alpinejs front-end mark-up in a classical laravel blade file. Inside that blade I have a dummy livewire component which acts as a bridge. The only downside is that we can't use the nice reactive updated public function in livewire3 backend as well as the frontend entangle. But event dispatching between Alpinejs parent and livewire childs is extremely simple and straightforward in livewire3.

Here is an example:

```
<x-app-layout>

<div 
            class="tab:padtop pt-5" 
            name="symbolsearch" 
            id="symbolsearch" 
         
            x-on:la-event-searchsymbol="symbols = $event.detail.symbols; openDropdown = symbols.length > 0"
            
            x-data="{

                    searchSymbol: '',
                
                    symbols: [], 
                    selectedSymbol: '',
                    tradeOpen: '',
                    
                    openDropdown: false,
                    open: false,

            }"

        >@livewire('dashboard.comp-symbol-search-single') 
        
            <div class="relative flex justify-center "> 

...Rest of reactive alpine mark-up
```

I transfer data from alpine as follows via $disptach:
```
 <input
                            x-ref="inputComponent" 
                            x-model.debounce.0ms="searchSymbol"
                            x-on:input="open = $event.target.value.length > 0; 
                            $dispatch('al-event-searchsymbol', { searchSymbol: $event.target.value });"     
...
```

In the livewire3 php backend I use the normal ways of disptaching and listening to events: 
```
$this->dispatch('la-event-searchsymbol', symbols: $this->symbols); 
```
And 
```
#[On('al-event-searchsymbol')] 
    public function updateSearchSymbol($searchSymbol)
```

The actual front-end livewire view is completely empty: 
```
<div>
    <!-- Diese Komponente dient nur der Livewire-Kommunikation -->
</div>
```

However I can not remove it entirely, because even when I add .window to the alpine x:on event listener I won't be able to listen to the livewire event. I suppose that without the @livewire('dashboard.comp-symbol-search-single')  livewire isn't actually used at all, so this line is needed despite being empty.

Now I have a full reactive alpinejs frontend outside of livewire and when I have a livewire update it transmits this html only :-) 
```
<div>
    <!-- Diese Komponente dient nur der Livewire-Kommunikation -->
</div>
```

**Comment by watee-dev** [comment]
Hello guys,

Did you solve this issue? I am facing the same issue after upgrade from livewire 2 to livewire 3. When I use wire:ignore on any children elemnt in livewire component, after update the whole DOM is updated and wire:ignore is skipped. I have a Vue 2 component which i need to wire:ignore. Vue component is wrapped by div element on ehich is applied wire:ignore. It doesnt work…
Thanks

**Comment by clementgonin** [comment]
Hi !

I've been having the same problem with wire:ignore not "ignoring", but strangely, only on my production environment, not on my local development environment. I'm using it to keep a CK Editor instance initialized, or I'd have to re-initialize it on each submit. 
I placed the wire:ignore on the container div around the <textarea> used for CK Editor, as recommanded in the doc.

After a few hours adding logs everywhere to try to understand what happened on the prod, I finally understood what made the difference...
On the production env, I'm caching views with the "optimize" command, and THAT'S what breaks wire:ignore in my case !

If I cache views on my local env, I get the same problem, with 2 errors in the console : 

```
Uncaught Snapshot missing on Livewire component with id: EBBLWuu16Brqhnzfhz6D
Uncaught Could not find Livewire component in DOM tree
```

I doubt this was the case for everyone having the problem in this thread, but well... 
Maybe it could help track down the problem ?

For now, I settled for not caching views on the production env (you can clear the views cache with "php artisan clear:views").
But if anyone has an idea why caching the views breaks wire:ignore... I'd be happy to be able to cache the views again :)
